java/core/src/main/java/co/worklytics/psoxy/gateway/impl/CommonRequestHandler.java

```diff
@@ -244,8 +210,9 @@ public URL buildTarget(HttpEventRequest request) {
         URIBuilder uriBuilder = new URIBuilder();
         uriBuilder.setScheme("https");
         uriBuilder.setHost(config.getConfigPropertyOrError(ProxyConfigProperty.TARGET_HOST));
-        uriBuilder.setPath(request.getPath());
-        uriBuilder.setCustomQuery(request.getQuery().orElse(""));
+        // URL comes encoded, decode it prior to perform call to API origin to avoid double encoding issues
+        uriBuilder.setPath(URLDecoder.decode(request.getPath(), StandardCharsets.UTF_8));
+        uriBuilder.setCustomQuery(URLDecoder.decode(request.getQuery().orElse(""), StandardCharsets.UTF_8));
```

Comment:
Double encoding fix

Reply:
actually, is this the right place??  I think it should be done within `HttpEVentRequest::getPath()`. Seems like parsing request --> nice POJO is responsibility of that class rather than the `CommonRequestHandler`

My particular concern is that GCP, AWS implementations each transform (or decorate) their native request format --> our generic `HttpEventRequest`.  Do they BOTH get double-encoded path values, such that this decoding is needed in both cases? or should only AWS's impl of `HttpEventRequest` have it?

Reply:
I do not agree, this is transversal to all. "=" in this particular case, like any other reserved character in the URL must travel encoded. Is not something that the AWS Api Gateway does, but standard behavior per [RFC](https://www.ietf.org/rfc/rfc1738.txt).
It is handler responsibility to know what to do with the URL, not each implementation to choose if decode or not.

Reply:
To be clear, up to this point the URL is perfectly valir. Problem is that we do another request with it, so it double encodes it.
We need to set it back to their "true self" before encoding, so the underlying transport takes care of the final encoding.

Reply:
OK, I guess I'm confused.  For some reason, I thought the behavior differs by platform , which is why we didn't hit this bug with GCP but do now with AWS. Specifically, if request is

`your-psoxy-host/gmail/v1/messages?pageToken=123123` then the POJO passed in by GCP looks like:

```json
{
  "path": "gmail/v1/messages",
  "query": "pageToken=123123"
}
```

We should be just re-assumably that into `gmail/v1/messages?pageToken=123123" to 1) test with the REGEX as to whether it's allowed by our rules and then 2) forward on to source. 

Your change suggests `getQuery()` returns `pageToken%3D123412`, whereas I expect it to return `pageToken=123123`.

Am I misunderstanding something?

Reply:
see
![image](https://user-images.githubusercontent.com/102831/152892175-6a1ffbcf-9dc9-42b0-923f-63b16c41c399.png)

Reply:
No, the problem comes when there are `=` in the parameter value. Page token is base64 so can have padding in the form `abc123=`

that when doing the call gets encoded to
`your-psoxy-host/gmail/v1/messages?pageToken=abc123%3D`

if we let the values as they come, the call that the proxy does to the endpoint will be re-encoded to
`your-psoxy-host/gmail/v1/messages?pageToken=abc123%253D`

so before that happens we need to set the token back to `abc123=`

Reply:
mmm.... AWS docs suggest they aren't double-encoding either:

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

so is this something we're doing on the Worklytics side? If so, I don't think it's the right thing. We want psoxy to be drop-in replacement for the data source's REST API. if our request to psoxy looks like `gmail/v1/messages?pageToken%3D123123` that is wrong

Reply:
Again, this is not anything the Api Gateway does. Is the underlying URLFetchService encoding the url.
When we do call directly GDirectory the call goes like this `your-psoxy-host/gmail/v1/messages?pageToken=abc123%3D`
That's perfectly valid and the way to go, the `=` needs to be encoded.
The thing is that we are chaining calls, our lambda does ANOTHER call to the api, so that's where the double encoding may happen-